audioutils/fmsynth/test/fmsynth_alsa_test.c

```diff
@@ -0,0 +1,324 @@
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <stdint.h>
+#include <termios.h>
+#include <fcntl.h>
+
+#include <alsa/asoundlib.h>
+
+#include <audioutils/fmsynth_eg.h>
+#include <audioutils/fmsynth_op.h>
+#include <audioutils/fmsynth.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define FS (48000)
+
+#define CHANNEL_NUM (2)
+#define RESAMPLING_ALSA (1)
+#define LATENCY_ALSA (10000)
+
+#define SAMPLE_NUM (FS / 20)
+#define BUFF_LENGTH (SAMPLE_NUM * CHANNEL_NUM)
+
+#define CODE_C_FREQ (261.625565f)
+#define CODE_D_FREQ (293.6647674f)
+#define CODE_E_FREQ (329.6275561f)
+#define CODE_F_FREQ (349.2282305f)
+#define CODE_G_FREQ (391.9954347f)
+#define CODE_A_FREQ (440.f)
+#define CODE_B_FREQ (493.8833009f)
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static int16_t samples[BUFF_LENGTH];
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * name: init_alsa
+ ****************************************************************************/
+
+static snd_pcm_t *init_alsa(int fs)
+{
+  int ret;
+  snd_pcm_t *hndl = NULL;
+
+  ret = snd_pcm_open(&hndl, "default", SND_PCM_STREAM_PLAYBACK, 0);
+  if (ret < 0)
+    {
+      printf("sdn_pcm_open error\n");
+      return NULL;
+    }
+
+  ret = snd_pcm_set_params(hndl,
+                           SND_PCM_FORMAT_S16,
+                           SND_PCM_ACCESS_RW_INTERLEAVED,
+                           CHANNEL_NUM, fs, RESAMPLING_ALSA, LATENCY_ALSA);
+  if (ret != 0)
+    {
+      printf("sdn_pcm_set_params error\n");
+      snd_pcm_close(hndl);
+      return NULL;
+    }
+
+  return hndl;
+}
+
+/****************************************************************************
+ * name: set_nonblocking
+ ****************************************************************************/
+
+static int set_nonblocking(struct termios *saved)
```

Comment:
```suggestion
static int set_nonblocking(FAR struct termios *saved)
```
Here and other similar places

Reply:
This code is just for unit testing on linux.
So I don't think it needs "FAR".